workflow/type.go

```diff
@@ -1,10 +1,20 @@
 package workflow
 
-import "github.com/mesg-foundation/engine/hash"
+import (
+	"github.com/mesg-foundation/engine/event"
+	"github.com/mesg-foundation/engine/hash"
+)
 
 // These structs are temporary and will be part of the service definition
 // TODO: Move to service struct
 
+type predicate uint
```

Comment:
```suggestion
type operator uint
```

and then EQ rename OP_EQ as a starndard notation

Reply:
Is `OP_EQ` really standard? It doesn't follow the camelCase naming rule

Reply:
I vote to keep the name `predicate` and to use explicit names closer to SQL:
```
equals (=)
greater than (>)
less than (<)
not equal to (<>)
less than or equal to (<=)
greater than or equal to (>=)
```

We can keep `operator` for `AND`, `OR` and `NOT` operators than can combine predicates.

http://www.cs.toronto.edu/~nn/csc309-20085/guide/pointbase/docs/html/htmlfiles/dev_searchcondandpredicates.html
https://en.wikipedia.org/wiki/Where_(SQL)
https://docs.microsoft.com/en-us/sql/t-sql/queries/search-condition-transact-sql?view=sql-server-2017

Reply:
or otherwise, something closer to noSQL:

Name | Description
-- | --
$eq | Matches values that are equal to a specified value.
$gt | Matches values that are greater than a specified value.
$gte | Matches values that are greater than or equal to a specified value.
$in | Matches any of the values specified in an array.
$lt | Matches values that are less than a specified value.
$lte | Matches values that are less than or equal to a specified value.
$ne | Matches all values that are not equal to a specified value.
$nin | Matches none of the values specified in an array.

https://docs.mongodb.com/manual/reference/operator/query/

Reply:
I'm ok to keep operator for AND and OR it makes more sense.

For the naming of the predicates, I like the simplicity of noSQL without the $ so EQ, GT, GTE...

Reply:
> Is OP_EQ really standard? It doesn't follow the camelCase naming rule

I can find more but here is the first example from golang source code
https://github.com/golang/go/blob/master/src/go/token/token.go

If you write some tokenizer/parser/lexer you don't use camelCase and this is one of the exceptions.

> We can keep operator for AND, OR and NOT operators than can combine predicates.

As you post [here](http://www.cs.toronto.edu/~nn/csc309-20085/guide/pointbase/docs/html/htmlfiles/dev_searchcondandpredicates.html)
predicates are more complex, including BETWEEN, LIKE function etc...:

```
PointBase supports the following types of predicates:

comparison (=, <>, <, >, <=, >=, !=)
BETWEEN
LIKE
EXITS | NOT EXISTS
IN | NOT IN
NULL
//....
```

but AND (&), OR (|) EQ(==), GT(>=) etc are all operators and I don't see the point to split AND, OR, NOT operators and EQ, ...

Also for the first implementation of workflow (there is no info about more complex syntax), we need only simple matching and there is no need to overengineering this part of the code at this point. We can start with operators because we need only them by now.

The question is where we want to go next. You want some DSL lang for it, or relay only on configuration syntax.